lib/virt_autotest/domain_management_utils.pm

```diff
@@ -274,4 +277,92 @@ sub check_guest_state {
+    my %guest_matrix = ();
+    foreach (0 .. scalar(split(/ /, $args{guest})) - 1) {
+        $guest_matrix{(split(/ /, $args{guest}))[$_]}{ipaddr} = (split(/ /, $args{ipaddr}))[$_];
+        $guest_matrix{(split(/ /, $args{guest}))[$_]}{domainname} = (split(/ /, $args{domainname}))[$_];
+    }
+    foreach (keys %guest_matrix) {
+        my $temp = 1;
+        my $hostname = $_ . '.' . $guest_matrix{$_}{domainname};
+        if ($args{usedns}) {
+            if (script_run("timeout --kill-after=1 --signal=9 15 ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i $args{keyfile} root\@$guest_matrix{$_}{ipaddr} ls") == 0) {
+                $temp = assert_script_run("timeout --kill-after=1 --signal=9 15 ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i $args{keyfile} root\@$guest_matrix{$_}{ipaddr} \"echo -e $hostname > /etc/hostname\;hostnamectl hostname $hostname;sync\"");
+            }
+            else {
+                enter_cmd("clear", wait_still_screen => 3);
+                enter_cmd("timeout --kill-after=1 --signal=9 30 ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i $args{keyfile} root\@$guest_matrix{$_}{ipaddr} \"echo -e $hostname > /etc/hostname\;hostnamectl hostname $hostname;sync\"", wait_still_screen => 3);
+                $temp = (check_screen("password-prompt", 60) ? 0 : 1);
+                enter_cmd(get_var('_SECRET_GUEST_PASSWORD', ''), wait_screen_change => 50, max_interval => 1);
+                $temp |= (wait_still_screen(35) ? 0 : 1);
+            }
+        }
+        else {
+            $temp = add_guest_to_hosts($_, $guest_matrix{$_}{ipaddr});
+        }
+        save_screenshot;
+        record_info("Guest $_ register name failed", "Either setting hostname or writing /etc/hosts failed", result => 'fail') if ($temp != 0);
+        $ret |= $temp;
+    }
+    return $ret;
+}
+
+=head2 manage_guest_service
+
+Manage service/target/socket unit in guest by using systemctl command. Multiple
+guests and corresponding ip addresses can be passed in as strings separated by
+space. Other arguments include keyfile (key file for passwordless ssh connection
+), operation (systemctl subcommand) and unit to be manipulated.
+=cut
+
+sub manage_guest_service {
+    my (%args) = @_;
+    $args{guest} //= '';
+    $args{ipaddr} //= '';
+    $args{keyfile} //= '/root/.ssh/id_rsa';
+    $args{operation} //= 'status';
+    $args{unit} //= 'default.target';
+    croak("Guest and ip aaddress must be given") if (!$args{guest} or !$args{ipaddr});
+
+    my $ret = 0;
+    my %guest_matrix = ();
+    $guest_matrix{(split(/ /, $args{guest}))[$_]}{ipaddr} = (split(/ /, $args{ipaddr}))[$_] foreach (0 .. scalar(split(/ /, $args{guest})) - 1);
+    foreach (keys %guest_matrix) {
+        my $temp = 1;
+        if (script_run("timeout --kill-after=1 --signal=9 15 ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i $args{keyfile} root\@$guest_matrix{$_}{ipaddr} ls") == 0) {
```

Comment:
How about putting ` ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i $args{keyfile}` to a var? Then the lines will be shorter.